### DISCUSSION TITLE
Developing iOS App Without iPhone

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10008] [General]
I'm using Arch-Linux to develop .NET apps. For Android, the default Android emulator "works" but is extremely slow, but installing Android-x86 in KVM works.

Now for iPhone... there is no VM version of iOS so that's not an option. Is there any way to test the app for iPhone without having to purchase an iPhone for that sole purpose?

Also it's impossible to deploy to the Apple Store and have an Apple Developer Account without having a MacBook.

It's just that... purchasing an iPhone, a MacBook, and then a $99/year account is a bit of a hefty price to deploy a small app.

**Comment by emmauss** [comment]
> I'm using Arch-Linux to develop .NET apps. For Android, the default Android emulator "works" but is extremely slow, but installing Android-x86 in KVM works.
> 
> Now for iPhone... there is no VM version of iOS so that's not an option. Is there any way to test the app for iPhone without having to purchase an iPhone for that sole purpose?
> 
> Also it's impossible to deploy to the Apple Store and have an Apple Developer Account without having a MacBook.
> 
> It's just that... purchasing an iPhone, a MacBook, and then a $99/year account is a bit of a hefty price to deploy a small app.

Unfortunately, it's not possible to develop for iOS without a Mac. It's possible to use the simulator on Mac if a real iOS device isn't available, but a Mac is required either way. This isn't a limitation we can work around on our end.

**Reply by mysteryx93** [reply]
So an iOS emulator is available on Mac, meaning only a MacBook is necessary but not an iPhone? What are the cheapest MacBook models that are suitable for his kind of work?

**Reply by maxkatz6** [reply]
> What are the cheapest MacBook models that are suitable for his kind of work?

Depending on the .NET SDK version is it has different requirements for the XCode. 
So having this https://devblogs.microsoft.com/dotnet/dotnet-maui-xcode-14/
And this https://developer.apple.com/forums/thread/708896
You need at least macOS Monterey 12.4 for the latest SDK experience.
Which means you have list of these models https://support.apple.com/en-us/HT212551

But, aside from that, my personal recommendation would be to look the any m1/apple silicon model, as you might want to build m1 application at some point (Avalonia and .NET both support it). Which limits your choice with only models from last ~2 years. To be specifically clear, you can build m1 apps only on m1 processor, as Apple requires it. Gladly, you can build x86 apps on m1 too.

**Reply by maxkatz6** [reply]
Technically, you can have mac machine on the cloud and connect from Windows Visual Studio to it. But I have a very little luck using this approach, it wasn't stable before. Not sure how is it now though.

**Reply by maxkatz6** [reply]
Btw, my first link to the MS blog also shows a little demo, how you can develop iOS app on Windows. But you still need a real mac device just connected and paired to the windows.

**Reply by mysteryx93** [reply]
To get the Apple Developer account, someone else I was thinking is to use the computer of a friend via remote control to create the account and for XCode. Any downsides I should be aware of with this approach? What happens if that friend disappears or dies, or the computer breaks or gets stolen? App income go to that account?

With Avalonia, the same app runs on both Android and iOS, so I can develop on Android and just take for granted that it also runs on iOS with very little testing... it gets a bit more tricky if I need mac-specific code that requires intensive testing.

And also... if I take my free apps like 432hz Player and 432hz Batch Converter, to publish on iOS store and Android store, let's say $5 each, got any idea how much income I can expect from that?

**Reply by maxkatz6** [reply]
> To get the Apple Developer account, someone else I was thinking is to use the computer of a friend via remote control to create the account and for XCode. Any downsides I should be aware of with this approach? What happens if that friend disappears or dies, or the computer breaks or gets stolen? App income go to that account?

If you own the account, there are no difference on which machine xcode is installed and app is build. In case of any problems you will be able to use your apple developer account on other machines. AFAIK there is no link between apple dev account used in xcode and mac local user.

> With Avalonia, the same app runs on both Android and iOS, so I can develop on Android and just take for granted that it also runs on iOS with very little testing... it gets a bit more tricky if I need mac-specific code that requires intensive testing.

Yeah, I can understand that. Was developing Xbox app on UWP for multiple years without even having an Xbox... 
Microsoft Xbox app-design guidelines documentation was my big friend back then and complaining users also helped a lot.

> And also... if I take my free apps like 432hz Player and 432hz Batch Converter, to publish on iOS store and Android store, let's say $5 each, got any idea how much income I can expect from that?

Can't help much with that. Assuming that iOS devices are primarily popular in NA/JP regions and less in EU, you can expect that $5 is fair there. In the rest of the world and Android in general prices are usually tented to be lower in comparison.
Regional pricing policy is hard.

**Comment by mysteryx93** [comment]
How about Remote iOS Simulator for Windows?
https://learn.microsoft.com/en-us/dotnet/maui/ios/remote-simulator?view=net-maui-7.0

**Reply by maxkatz6** [reply]
> In Visual Studio 2022, pair the IDE to a Mac Build host if you haven't previously. For more information, see [Pair to Mac for iOS development](https://learn.microsoft.com/en-us/dotnet/maui/ios/pair-to-mac?view=net-maui-7.0).

You still need a mac to pair with windows. It might be a device from the cloud.

**Comment by mysteryx93** [comment]
Hey... can I run OS X via KVM; and then iOS emulator from OS X?
https://github.com/foxlet/macOS-Simple-KVM

Only downside is that I won't link directly to iOS from Linux, but would have to run the project from OS X. Can install JetBrains Rider in OS X no problem.

Other benefit is that I get to test on Mac OS.

**Reply by maxkatz6** [reply]
I heard it's possible to run Xcode on it, but never tried. No idea if it will work with emulator.

**Reply by mysteryx93** [reply]
Trying to expose USB debugging from QEMU Android-x86 to the host... that seems pretty complicated with no clear instructions for Arch Linux.

Since it's about setting up USB bridges... technically it might be possible to pass the bridge from iOS emulator to MacOS to Linux? But that would probably be highly complicated and require deep knowledge of networking and emulators both on Linux and Mac... easier to install Rider in Mac.

**Reply by kekekeks** [reply]
1) Rider doesn't work well on virtualized macs without physical GPU passthrough.
2) for android you don't need USB debugging, adb can work over TCP/IP.

**Comment by Saladin1812** [comment]
Something like expo go would help alot.. the only reason react native / expo is really good for me is the fact that I can develop from linux and test on my IOS device.
